core/repository/delegate.ts

```diff
@@ -1,68 +1,41 @@
 import Response from 'shared/model/response';
-const crypto = require('crypto');
-import { AccountRepository } from 'shared/repository/account';
 import { Account } from 'shared/model/account';
-import { ed } from 'shared/util/ed';
-
-interface IForgingDisable {
-    secret: string;
-    publicKey: string;
-}
-
-export interface IDelegateRepository {
-    forgingDisable(data: IForgingDisable): Promise<Response<any>>;
-
-    forgingEnable(data: IForgingDisable): Promise<Response<any>>;
-}
-
-class DelegateRepository implements IDelegateRepository {
-    private accountRepo: AccountRepository;
-
-    constructor() {
-        this.accountRepo = new AccountRepository();
-    }
-
-    async forgingDisable(data: IForgingDisable): Promise<Response<any>> {
-        const hash = crypto.createHash('sha256').update(data.secret, 'utf8').digest();
-        const publicKey: string = ed.makePublicKeyHex(hash);
-        const account  = await this.accountRepo.getAccount(publicKey);
-        if (!account) {
-            return new Response({
-                errors: ['Delegate not found']
-            });
-        }
-        return new Response({
-            data: {
-                address: account.data.address
+import Config from 'shared/util/config';
+import { IDelegate } from 'shared/model/delegate';
+
+const constants = Config.constants;
+
+class DelegateRepository {
+    private memoryDelegates: Array<IDelegate> = [];
```

Comment:
For quick access, you need to store delegates in an object:
memoryDelegates: { [address: number]: IDelegate } = {};